jms/src/main/scala/akka/stream/alpakka/jms/Exceptions.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.jms
+
+/**
+ * Marker trait indicating that the exception thrown is intermittent. The failed operation might succeed if tried again.
+ */
+trait RetriableException extends Exception
+
+/**
+ * Marker trait indicating that the exception thrown is persistent. The operation will always fail when retried.
+ */
+trait NonRetriableException extends Exception
```

Comment:
Since this is a jms library, maybe we should call this a `NonRetriableJmsException`.